src/design-system/components/input-field/InputField.tsx

```diff
@@ -0,0 +1,81 @@
+import '~scss-components/input-field/InputField.scss'
+import { InputFieldVariantEnum } from '~/types'
+import SearchIcon from '@mui/icons-material/Search'
+import ClearIcon from '@mui/icons-material/Clear'
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline'
+import { cn } from '~/utils/cn'
+
+import { InputBaseProps } from '@mui/material/InputBase'
+
+import Box from '@mui/material/Box'
+import { SxProps } from '@mui/material'
+
+export interface InputFieldProps extends InputBaseProps {
+  variant: InputFieldVariantEnum
+  label?: string
+  disabled?: boolean
+  value: string
+  placeholder: string
+  helperText?: string
+  search?: boolean
+  error?: boolean
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
+}
+
+const InputField: React.FC<InputFieldProps> = ({
+  variant = InputFieldVariantEnum.Small,
+  label,
+  disabled,
+  value,
+  placeholder,
+  helperText,
+  error,
+  search,
+  onChange,
+  sx
+}) => {
+  const clearInput = () => {
+    onChange({ target: { value: '' } } as React.ChangeEvent<HTMLInputElement>)
+  }
+
+  return (
+    <Box sx={sx as SxProps}>
+      <div
+        className={cn('s2s-input-container', `s2s-input-container_${variant}`, {
+          's2s-input-container_disabled': disabled,
+          's2s-input-container_error': error
+        })}
```

Comment:
I think using `cn` this way (without object) will be more readable and simple, but it's up to you
```suggestion
        className={cn('s2s-input-container', 
        `s2s-input-container_${variant}`,
        disabled && 's2s-input-container_disabled',
        error && 's2s-input-container_error'
        )}
```